File: app/src/main/java/com/mpt/android/multitextwatcherdemo/MainActivity.java
package com.mpt.android.multitextwatcherdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.Editable;
import android.util.Log;
import android.widget.EditText;

import com.mpt.android.multitextwatcher.MultiTextWatcher;
import com.mpt.android.multitextwatcher.OnTextWatcher;

public class MainActivity extends AppCompatActivity implements OnTextWatcher {


    private static final String TAG = "MultiTextWatcher";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText etName = (EditText) findViewById(R.id.etMainName);
        EditText etEmail = (EditText) findViewById(R.id.etMainEmail);
        EditText etPassword = (EditText) findViewById(R.id.etMainPassword);
        EditText etPhone = (EditText) findViewById(R.id.etMainPhone);

        new MultiTextWatcher()
                .setEditText(etEmail)
                .setOnTextWatcher(this);
        new MultiTextWatcher()
                .setEditText(etName)
                .setOnTextWatcher(this);
        new MultiTextWatcher()
                .setEditText(etPassword)
                .setOnTextWatcher(this);
        new MultiTextWatcher()
                .setEditText(etPhone)
                .setOnTextWatcher(this);


    }

    @Override
    public void beforeTextChanged(EditText editText, CharSequence s, int start, int count, int after) {


        switch (editText.getId()) {
            case R.id.etMainEmail:
                // TODO: your task
                break;

        }

        Log.i(TAG, "beforeTextChanged: " + s + "Hint--" + editText.getHint());
    }

    @Override
    public void onTextChanged(EditText editText, CharSequence s, int start, int before, int count) {

        switch (editText.getId()) {
            case R.id.etMainEmail:
                // TODO: your task
                break;

        }
        Log.i(TAG, "onTextChanged: " + s + "Hint--" + editText.getHint());
    }

    @Override
    public void afterTextChanged(EditText editText, Editable editable) {

        switch (editText.getId()) {
            case R.id.etMainEmail:
                // TODO: your task
                break;

        }
        Log.i(TAG, "afterTextChanged: " + editable.toString() + "Hint--" + editText.getHint());
    }
}

File: app/src/main/java/com/mpt/android/multitextwatcherdemo/MainActivity.kt
package com.mpt.android.multitextwatcherdemo

import android.os.Bundle
import android.text.Editable
import android.util.Log
import android.view.View
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.mpt.android.multitextwatcher.MultiTextWatcher
import com.mpt.android.multitextwatcher.OnTextWatcher


class MainActivity : AppCompatActivity(), OnTextWatcher {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val etName: EditText = findViewById<View>(R.id.etMainName) as EditText
        val etEmail: EditText = findViewById<View>(R.id.etMainEmail) as EditText
        val etPassword: EditText = findViewById<View>(R.id.etMainPassword) as EditText
        val etPhone: EditText = findViewById<View>(R.id.etMainPhone) as EditText


        MultiTextWatcher(etName).setOnTextWatcher(this)
        MultiTextWatcher(etEmail).setOnTextWatcher(this)
        MultiTextWatcher(etPassword).setOnTextWatcher(this)
        MultiTextWatcher(etPhone).setOnTextWatcher(this)
    }

    override fun beforeTextChanged(
        editText: EditText,
        s: CharSequence,
        start: Int,
        count: Int,
        after: Int
    ) {
        when (editText.id) {
            R.id.etMainEmail -> {
            }
        }
        Log.i(TAG, "beforeTextChanged: " + s.toString() + "Hint--" + editText.hint)
    }

    override fun onTextChanged(
        editText: EditText,
        s: CharSequence,
        start: Int,
        before: Int,
        count: Int
    ) {
        TODO("Not yet implemented")
    }

    override fun afterTextChanged(editText: EditText, editable: Editable) {
        TODO("Not yet implemented")
    }


    companion object{
        private const val TAG: String = "MultiTextWatcher"
    }
}